pepper-apis/src/main/java/org/broadinstitute/ddp/export/DataExporter.java

```diff
@@ -219,6 +219,41 @@ public DataExporter(Config cfg) {
         return activities;
     }
 
+    /**
+     * Find and set the max number of instances seen per participant across the study for each given activity extract.
+     *
+     * @param handle     the database handle
+     * @param activities the list of activities to look at
+     */
+    public void computeMaxInstancesSeen(Handle handle, List<ActivityExtract> activities) {
+        var instanceDao = handle.attach(ActivityInstanceDao.class);
+        for (var activity : activities) {
+            long activityId = activity.getDefinition().getActivityId();
+            long versionId = activity.getVersionDto().getId();
+            int maxInstancesSeen = instanceDao
+                    .findMaxInstancesSeenPerUserByActivityAndVersion(activityId, versionId)
+                    .orElse(0);
+            activity.setMaxInstancesSeen(maxInstancesSeen);
+        }
+    }
+
+    /**
+     * Find and set the attribute names seen across all participants in study for each given activity extract.
+     *
+     * @param handle     the database handle
+     * @param activities the list of activities to look at
+     */
+    public void computeActivityAttributesSeen(Handle handle, List<ActivityExtract> activities) {
+        var instanceDao = handle.attach(ActivityInstanceDao.class);
+        for (var activity : activities) {
+            long activityId = activity.getDefinition().getActivityId();
```

Comment:
Showing the type of `activity` or `activities`  or both would be helpful here.

Reply:
Done